Add Data Warehouse docs (#6156)
PostHog/posthog.com: contents/docs/data/data-warehouse.mdx

---
title: Data Warehouse
sidebar: Docs
showTitle: true
availability:
    free: full
    selfServe: full
    enterprise: full
---

> The PostHog Data Warehouse gives you a place to put all of your most important data and query across these datasets, combined with the product analytics data already in PostHog. Powered by Clickhouse, the Data Warehouse allows you to query across big data sets quickly. 

Reviewer: ```suggestion
The PostHog Data Warehouse gives you a place to put all of your most important data. Powered by Clickhouse, the Data Warehouse allows you to query across these big data sets quickly. 
```

Data Warehouse is currently in **private beta**. This means we're inviting a select number of people to try this product out.

## How does it work?

> Each table you create is backed by one or more files in your object storage system (ie S3, GCS). We never "ingest" your data into our systems. Instead, on every query, we'll read directly from your s3 bucket. This means you can query as much data as you want. We'll only charge you for the data we actually query.

Reviewer: nice explanation

> Each table you create is backed by one or more files in your object storage system (ie S3, GCS). We never "ingest" your data into our systems. Instead, on every query, we'll read directly from your s3 bucket. This means you can query as much data as you want. We'll only charge you for the data we actually query.

Reviewer 2: ```suggestion
Each table in the warehouse is backed by one or more files in your object storage system (i.e. S3, GCS). We never "ingest" your data into our systems. Instead, on every query, we'll read directly from your object storage system. This means you can query as much data as you want. We'll only charge you for the data we actually query.
```

> > Data Warehouse is free to use while in private beta. After that, you'll be charged for each GB of data read while querying.

Reviewer 2: ```suggestion
> Data warehouse is free to use while in private beta. After that, you'll be charged for each GB of data read while querying.
```

## Creating a table

To create a new table, you'll need to:

> 1. Create a bucket

Reviewer 2: ```suggestion
1. Create a bucket in your object storage system like S3 or GCS
```

2. Set up an access key and secret
3. Add data to the bucket (we'll use Airbyte)
4. Create the table in PostHog

> These docs are written for AWS S3, but you can also use Google Cloud Storage (GCS).

> ### Step 1: Creating a bucket

Reviewer 2: ```suggestion
### Step 1: Creating a bucket in S3
```

1. Log in to [AWS](https://console.aws.amazon.com/).
2. Open [S3](https://s3.console.aws.amazon.com/) in the AWS console and create a new bucket. We suggest `us-east-1` if you're using PostHog Cloud US, or `eu-central-1` if you're using PostHog Cloud EU.

![created bucket](../../images/docs/apps/s3-export/bucket.png)

> Make sure to take note of both the name of your bucket and the region that it's in, we'll need these later.

### Step 2: Set up access

> Next, we'll need to create a new user in our AWS console with programmatic access to our newly created bucket.

Reviewer 2: ```suggestion
Next, we need to create a new user in our AWS console with programmatic access to our newly created bucket.
```

> 1. Open [IAM](https://console.aws.amazon.com/iam/home) and create a new policy to allow access to this bucket.

Reviewer 2: ```suggestion
1. Open [IAM](https://console.aws.amazon.com/iam/home) and create a new policy to enable access to this bucket
```

> 2. On the left under "Access management" select "Policies" and click "Create policy"

Reviewer 2: ```suggestion
2. On the left under "Access management," select "Policies," and click "Create policy"
```

3. Under the service, choose "S3"

> 4. Under "Actions" select

Reviewer 2: ```suggestion
4. Under "Actions," select:
```

### Step 3: Add data to the bucket

> For this section, we'll be using **Airbyte**. However, we accept any data in CSV or Parquet format, so if you already have data in S3 you can skip this section.

1. Go to [Airbyte](https://airbyte.com) and sign up for an account if you haven't already.
2. Go to connections and click "New connection"
3. Select a source. For this example we'll be grabbing data from **Stripe**, but you can use any of Airbyte's sources.
4. Click "Set up a new destination"
5. Select "S3" as the destination
6. Fill in the "S3 Bucket Name", "S3 Bucket Region" with the name and region you created earlier.
7. For "S3 Bucket Path", use `airbyte`.
8. For the "Output Format", pick Parquet. You can use the default settings
9. Under "Optional fields", you'll want to add the access key and secret from step 1.
10. In the next step, pick the streams you want to fill. Given you'll manually need to create a table for each stream, we suggest being selective.
11. Wait for the sync to finish

![creating the s3 destination](../../images/features/data-warehouse/airbyte-destination.png)

### Step 4: Create the table

> 1. In PostHog, go to the "Data Warehouse" tab and click "New table"

Author: Do we always link to app, and assume people figure out EU?

> 1. In PostHog, go to the "Data Warehouse" tab and click "New table"

Reviewer 2: It automatically redirects to EU if they are logged in there

2. Fill in the table name.

> 3. For the URL pattern, copy the URL from s3, up to the bucket name. Replace the file name with `*`, as Airbyte will split larges streams out into multiple files.

Reviewer 2: ```suggestion
3. For the URL pattern, copy the URL from S3, up to the bucket name. Replace the file name with `*`, as Airbyte will split larges streams out into multiple files.
```

  - For example: `https://your-bucket.s3.amazonaws.com/airbyte/invoices/*`
4. For file format, select Parquet. Fill in the access key and secret key.

> You'll want to repeat this step for each "stream" or folder that Airbyte created in your S3 bucket.

![creating table in posthog](../../images/features/data-warehouse/create-table.png)

### Step 5: Query the table.

> Amazing! You can now query your new table in any HogQL query.

Reviewer 2: ```suggestion
Amazing! You can now query your new table in any [HogQL query](/docs/product-analytics/hogql).
```
1. An example would be nice. Maybe a code block with an SQL insight that lists transactions from Stripe.
2. Can users now find this data in the database tab of data management? If so, we should mention and link to there.
